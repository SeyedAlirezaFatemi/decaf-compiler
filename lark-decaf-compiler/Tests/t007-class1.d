
class Person {
    string name;
    int age;

    void setName(string new_name) {
        name = new_name;
    }

    void setAge(int new_age) {
        age = new_age;
    }

    void print() {
        Print("Name: ", name, " Age: ", age);
    }

}

int main() {
    Person p;

    string name;
    int age;

    name = ReadLine();
    age = ReadInteger();

    p = new Person;
    p.setName(name);
    p.setAge(age);

    p.print();
}
