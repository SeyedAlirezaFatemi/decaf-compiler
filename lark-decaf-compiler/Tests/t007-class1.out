Name: Bertrand Russel Age: 98
